src/System.Collections.Immutable/src/Properties/SecurityTransparent.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security;
+
+[assembly: SecurityTransparent]
```

Comment:
I was referring to the general case. I'm OK with individual libraries adding this if the right testing and security audits are done (i.e. we know we don't call critical code without proper checks and annotations).